rotkehlchen/tests/unit/test_globaldb.py

```diff
@@ -358,3 +358,72 @@ def test_get_all_asset_data_specific_ids(globaldb):
         mapping=True,
         specific_ids=[],
     ) == {}
+
+
+def test_globaldb_pragma_foreign_keys(globaldb):
+    """
+    This tests verifies the behaviour of sqlite at the moment of
+    activating and deactivating PRAGMA foreign_keys. As per what
+    we could test and the documentation says for the change to
+    take effect it needs that no transaction is ongoing.
+
+    In this test we do:
+    - Deactivate the check, insert assets and activate it again
+    - Deactivate them and activate them again without pending transactions
+    - Start a transaction, check that the change doesn't take effect, commit
+    and see how it then works.
+    """
```

Comment:
Also at the end of the docstring add:

```suggestion

    The reason for this test's existence is to verify our assumptions on sqlite and PRAGMAs and to check that they hold for the versions we use. If this test fails we know that something has changed and we will need to adjust our strategy.
    """
```